src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,143 @@
+package frc.robot.subsystems.conveyor;
+
+import com.ctre.phoenix.motorcontrol.TalonFXInvertType;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.*;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Ports;
+
+import java.util.Deque;
+import java.util.LinkedList;
+
+public class Conveyor extends SubsystemBase {
+    private static Conveyor INSTANCE = null;
+    private final WPI_TalonFX motor = new WPI_TalonFX(Ports.Conveyor.MOTOR);
+    private final Deque<String> position = new LinkedList<>();
+    private final ColorSensorV3 colorSensorIntake = new ColorSensorV3(I2C.Port.kOnboard);
+    private final DigitalInput beamBreaker = new DigitalInput(Ports.Conveyor.BEAM_BREAKER);
+    private final Solenoid flap = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Conveyor.SOLENOID);
+    ColorMatch match = new ColorMatch();
+    private DriverStation.Alliance lastSeenColor = DriverStation.Alliance.Invalid;
+    private boolean lastPassed = true;
+
+    private Conveyor() {
+        motor.setInverted(TalonFXInvertType.Clockwise);
+        match.addColorMatch(Color.kRed); // red
+        match.addColorMatch(Color.kBlue); // blue
+        match.addColorMatch(Color.kGreen);
+        match.addColorMatch(Color.kYellow);
+        match.addColorMatch(Color.kBlack);

```

Comment:
Why do you need so many colors?

Reply:
Otherwise it returns blue and red when it shouldn't